AirCasting/SearchAndFollow/CompleteScreen/Views/CompleteScreenViewModel.swift

```diff
@@ -172,13 +172,23 @@ class CompleteScreenViewModel: ObservableObject {
                     if let stream = self.externalSessionWithStreams!.streams.first {
                         self.selectedStream = stream.id
                         self.selectedStreamUnitSymbol = stream.unitSymbol
-                        (self.chartStartTime, self.chartEndTime) = self.chartViewModel.generateEntries(with: stream.measurements.map({ SearchAndFollowChartViewModel.ChartMeasurement(value: $0.value, time: $0.time) }), thresholds: stream.thresholdsValues)
+                        (self.chartStartTime, self.chartEndTime) = self.chartViewModel.generateEntries(with: stream.measurements.map({ SearchAndFollowChartViewModel.ChartMeasurement(value: $0.value, time: $0.time) }), thresholds: stream.thresholdsValues, basedOn: self.componentsSeparation(name: stream.sensorName))
                     }
                 }
             }
         }
     }
     
+    private func componentsSeparation(name: String) -> String {
+        if name.contains(":") {
+            let value = name.components(separatedBy: ":").first!
+            return value.components(separatedBy: CharacterSet.decimalDigits).joined()
+        } else {
+            let value = name.components(separatedBy: "-").first!
```

Comment:
that's a bit dangerous. I believe that `stream.sensorName` is provided externally (so by the API or a user or smth we don't fully control). This is why we cannot make that assumption that if it doesn't contain `:` it will 100% sure contain `-`. I think we need to be prepared to fail here. Please implement a fallback or error handling